api/v1alpha1/etcdcluster_webhook.go

```diff
@@ -98,11 +102,42 @@ func (r *EtcdCluster) ValidateUpdate(old runtime.Object) (admission.Warnings, er
 		return warnings, err
 	}
 
-	return warnings, nil
+	return warnings, validateExtraArgs(r)
```

Comment:
I think we should use `allErrors` for possible errors returned from `validateExtraArgs` to return all validation errors at once

Reply:
Fixed. Now error looks like aggregate of errors:
Error from server (Forbidden): error when creating "config/samples/etcd.aenix.io_v1alpha1_etcdcluster.yaml": admission webhook "vetcdcluster.kb.io" denied the request: [Extra arg should not start with dash and have trailing dash, but it can contain dashes in the middle. flag: --xxx, can't use extra argument 'auto-tls' in .Spec.PodSpec.ExtraArgs because it generated by operator for cluster setup, can't use extra argument 'name' in .Spec.PodSpec.ExtraArgs because it generated by operator for cluster setup]

Reply:
Error on update:
# etcdclusters.etcd.aenix.io "etcdcluster-sample" was not valid:
# * spec.podSpec.extraArgs: Invalid value: map[string]string{"--xxx":"sdfdsf", "auto-tls":"", "name":"avkontya"}: [can't use extra argument 'auto-tls' in .Spec.PodSpec.ExtraArgs because it generated by operator for cluster setup, can't use extra argument 'name' in .Spec.PodSpec.ExtraArgs because it generated by operator for cluster setup, Extra arg should not start with dash and have trailing dash, but it can contain dashes in the middle. flag: --xxx]